### PR TITLE
Add proxy protocol config api message.

### PR DESCRIPTION
**Pull request by wez470**
Description: This PR creates a common PROXY protocol config API message. Currently it will be used for @alyssawilk's CONNECT work as well as in the transport socket for my upstream proxy proto work. This message could be extended to include TLVs in the future.
Risk Level: Low
Testing: None
Docs Changes: None
Release Notes: None
Discussed in: https://github.com/envoyproxy/envoy/pull/10682 (my draft PR to discuss the upstream implementation)

Wasn't 100% sure where the best place to stick this would be but I put it in types for now.

### CODE REVIEW

**Comment by wez470** [comment]
I think this build rule covers it https://github.com/envoyproxy/envoy/blob/master/api/envoy/type/v3/BUILD?

I _think_ I ran proto_format, but will double check after work today

**Comment by wez470** [comment]
Definitely forgot to add the shadow file when I committed :sweat_smile:

File: api/envoy/type/v3/proxy_protocol.proto [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.type.v3";
+option java_outer_classname = "ProxyProtocolProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Proxy Protocol]
+
+message ProxyProtocolConfig {
```
Comment (htuch) [inline api/envoy/type/v3/proxy_protocol.proto:15]: Is this a core type or does it belong somewhere like `envoy.config.core`? I prefer to keep the type hierarchy for fundamental types.

The reason that there is no v4 is that this does not reference any deprecated fields or other types that have transitively deprecated fields.

Comment (wez470) [reply api/envoy/type/v3/proxy_protocol.proto:15]: Yeah, I wasn't sure where to put it. Taking another look, the types already here probably would be more widely used than the proxy proto config. I can move to `envoy.config.core`

Comment (mattklein123) [reply api/envoy/type/v3/proxy_protocol.proto:15]: +1, agreed on moving it. Thank you.

/wait

**Comment by wez470** [comment]
@htuch @mattklein123 @alyssawilk ready for another look

**Pull request merged**